src/main/java/FigureSupplier.java

```diff
@@ -0,0 +1,71 @@
+import java.util.Random;
+
+public class FigureSupplier {
+    public static final int FIGURE_COUNT = 5;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        switch (figureNumber) {
+            case 1: return getRandomSquare();
+            case 2: return getRandomCircle();
+            case 3: return getRandomRectangle();
+            case 4: return getRandomIsoscelesTrapezoid();
+            case 5: return getRandomRightTriangle();
+            default: return getDefaultFigure();
+        }
+        // generate a specific figure based on the `figureNumber` value
+    }
+
+    private Figure getRandomSquare() {
+        Square square = new Square();
+        square.setSide(random.nextInt(10));
+        square.setColor(colorSupplier.getRandomColor());
+        square.getArea();
```

Comment:
why do we need to call this method here? We don't use it's return value nowhere (same in other methods)